packages/experimental/RadioGroup/src/Radio/useRadio.ts

```diff
@@ -0,0 +1,86 @@
+import { RadioProps, RadioState } from './Radio.types';
+import * as React from 'react';
+import { useRadioGroupContext } from '../RadioGroup/radioGroupContext';
+import { useAsPressable, useOnPressWithFocus, useViewCommandFocus } from '@fluentui-react-native/interactive-hooks';
+import { radioSelectActionLabel } from './Radio.styling';
+
+export const useRadio = (props: RadioProps): RadioState => {
+  const defaultComponentRef = React.useRef(null);
+  const {
+    value,
+    disabled,
+    accessibilityLabel,
+    componentRef = defaultComponentRef,
+    accessibilityPositionInSet,
+    accessibilitySetSize,
+    ...rest
+  } = props;
+
+  // Grabs the context information from RadioGroup (currently selected button and client's onChange callback)
+  const info = useRadioGroupContext();
+
+  const buttonRef = useViewCommandFocus(componentRef);
+
+  /* We don't want to call the user's onChange multiple times on the same selection. */
+  const changeSelection = React.useCallback(() => {
+    if (value != info.value) {
+      info.onChange && info.onChange(value);
+      info.updateSelectedButtonRef && componentRef && info.updateSelectedButtonRef(componentRef);
+    }
+  }, [info, value, componentRef]);
+
+  /* We use the componentRef of the currently selected button to maintain the default tabbable
+    element in a RadioGroup. Since the componentRef isn't generated until after initial render,
+    we must update it once here. */
+  React.useEffect(() => {
+    if (value == info.value) {
+      info.updateSelectedButtonRef && componentRef && info.updateSelectedButtonRef(componentRef);
+    }
+  }, [info, value, componentRef]);
```

Comment:
I feel like this would fire more often than it should because it's taking a dependency on the entire info object instead of the parts of the info object that matter, so any time anything on info is changed, info would be recreated and trigger this useEffect.

Reply:
Also this seems redundant except for the first time it runs because of line 28... Is it possible to have it only run after first render by passing [] as the dependency array? Since this would run after the first render the ref should be populated by the time it runs the first time...

Reply:
That makes sense! Yes, I can pass in an empty dependency array so that it only runs once at the start. Updated!